Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/setup.py

```diff
@@ -37,6 +38,8 @@ class Setup(Command):
     name = 'setup'
     help = 'Configure local settings for the current repository'
 
+    REMOTE_RE = re.compile(r'(?P<protcol>[^@:]+)(@|://)(?P<host>[^:]+)(/|:)(?P<path>\S+)(\.git)?/?')
```

Comment:
As in a previous comment, it would be nice if we could use `urllib.parse` or similar to parse git repo URLs rather than relying on a complex regex.